### DISCUSSION TITLE
CIDR(Classless Inter-Domain Routing, 사이더) 란?

### DISCUSSION THREAD

**Discussion by seunghye218** [2024-woowacourse-study/level-interview#207] [Network]
.

**Comment by Kimprodp** [comment]
CIDR(Classless Inter-Domain Routing)은 IP 주소 할당과 라우팅을 효율적으로 관리하기 위한 방법으로, 기존 네트워크 클래스로 나눠 정의하던 IP정보를 Class 없이 유연하게 나눠줄 수 있는 라우팅 기법이다.

원래는 Class로 A, B, C, D, E 클래스로 나누어서 네트워크를 사용했지만,
클래스로 나누게 되는 경우 IP들은 유연성이 없고 IP의 개수가 많기 때문에 좀 더 유연하게 사용하고자 등장했다.

CIDR 표기법에서는 IP 주소와 서브넷 마스크를 슬래시(/)로구분하여 표기한다.
- 예를 들어 192.168.1.0/24는 IP 주소 192.168.1.0를 나타냄
- 슬래시 뒤의 숫자는 서브넷 마스크의 1 비트 개수를 나타내며, 네트워크 부분과 호스트 부분을 구분
- /24 는 24비트의 네트워크 부분으로 255.255.255.0을 의미

**Comment by seunghye218** [comment]
## CIDR

CIDR(Classless Inter-Domain Routing)은 IP 주소를Class  A,B,C 등과 같이 규격화된 구분없이, 비트 단위의 서브넷 마스크 정보도 네트워크 정보로써 함께 라우팅하는 것입니다. IPv4의 `IP 주소 고갈` 및 `라우팅 테이블 대형화`에 대한 해소책으로 나왔습니다. (표준 :RFC 4632)

## Classful Routing, Classless Routing 비교

- Classful Routing: Class (A,B,C 등)별로 규격화된 IP주소 구분에 따라 나누어진 네트워크 주소 정보를 토대로 라우팅 하는 것
- Classless Routing: 고정길이가 아닌 가변 길이로 주소를 묶어주는 슈퍼네트마스크(슈퍼넷팅) 개념이다. 
  - 이 경우,라우팅 테이블이나 라우팅 정보 교환시 데이터량을 절약할 수 있다. 특히, CIDR은, 좌측 부터 비트 단위(IP 프리픽스)로 넷마스크(CIDR마스크)를 취하는 방식이다. 결국, 규격화된 C Class의 경우, 약 2백만개 이상의 주소 블록 범위가 있게 된다. 라우터가 이를 처리하려면, 라우터의 라우팅 테이블은 엄청난 메모리가 필요하게 된다. 이때, CIDR 처럼 프리픽스를 이용한 비트 단위의 주소 할당이 된다면, 계층적인 주소 할당 및 계층적 라우팅이 이루어지게 된다.
